src/server/debug/server/server.go

```diff
@@ -927,3 +952,123 @@ func redirectDump(ctx context.Context, c debug.DebugClient, filter *debug.Filter
 	}
 	return grpcutil.NewStreamingBytesReader(dumpC, nil), nil
 }
+
+func (s *debugServer) collectDatabaseStats(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	if err := s.collectDatabaseActivities(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseRowCount(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseTableSizes(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *debugServer) collectDatabaseActivities(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "activities", queryDatabaseActivities, prefix...)
+}
+
+func (s *debugServer) collectDatabaseRowCount(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "row-counts", queryDatabaseRowCounts, prefix...)
+}
+
+func (s *debugServer) collectDatabaseTableSizes(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "table-sizes", queryDatabaseTableSizes, prefix...)
+}
+
+func (s *debugServer) collectDatabaseQuery(ctx context.Context, tw *tar.Writer, fileName string,
+	queryFunc func(conn *sql.Conn, ctx context.Context, w io.Writer) error, prefix ...string) error {
+	return collectDebugFile(tw, fileName, "txt", func(w io.Writer) error {
+		conn, err := s.env.GetDBClient().Conn(ctx)
```

Comment:
Conn() does not limit the query's runtime to this context; it basically controls how long we'll block to establish or reuse a new connection.  I have never called this before and I don't think it's the right thing to do here.  with the DB object, you can `QueryContext(ctx, ...` or `QueryxContext(ctx, ...` to run the query with a timeout.

Reply:
Also, I think the database should be passed as a parameter to the server's constructor.  Makes it explicit that this will make database calls.  (s.env.GetDBClient() will never return a different object.)

Reply:
Originally I wanted to pass the conn down to make testing easier, but you make a good point. I'll update this as well